### PR TITLE
Add Jupyter extension [risk: low]

### LINKED ISSUE #491: Increase the usable area for notebooks
Multiple users have noted:
[When running notebooks] A lot of the screen is taken up Saturn and ipython banners leaving very little real estate for notebook code or outputs (plots)

Decreasing the size of the page footer seems like one obvious area to save room

### PR DESCRIPTION
**Pull request by zarsky-broad**
resolves #491 

~**Until Leo supports http url's for notebook extensions, this is non-functional and trying to open it will just regenerate all your clusters for nothing...**~

**Update:** should work on dev!

Caveats: Since this uses the window.location to figure out the script url to send to Leo, it can't be properly tested locally. As such, any regenerating of clusters by anyone on localhost won't have this correctly. We could fix this by adding the url of dev to the dev config.

### CODE REVIEW

File: src/pages/workspaces/workspace/notebooks/NotebookLauncher.js [hunk 1]
```diff
@@ -50,14 +50,21 @@ class NotebookLauncherContent extends Component {
...
       await Promise.all([
         this.localizeNotebook(clusterName),
         Jupyter.notebooks(namespace, clusterName).setCookie()
       ])
 
       const { name: workspaceName, notebookName } = this.props
-      this.setState({ url: `${clusterUrl}/notebooks/${workspaceName}/${notebookName}` })
+      this.setState({ url: `${clusterUrl}/notebooks/${workspaceName}/${notebookName}` },
+        () => {
+          window.addEventListener('message', e => {
```
Comment (panentheos) [inline src/pages/workspaces/workspace/notebooks/NotebookLauncher.js]: We should make sure this gets cleaned up on unmount. To go along with that, could we just add this immediately on mount for simplicity, or is there a reason we need to wait for this state transition?

**Review by panentheos — APPROVED** [review]
Looks good, see comment.

**Comment by zarsky-broad** [comment]
Oops, I knew there was something I meant to do yesterday that I forgot...

**Comment by zarsky-broad** [comment]
@panentheos Any thoughts about the caveat I listed in the description above?

**Comment by panentheos** [comment]
It sounds like we should have this pull from dev explicitly, otherwise the local cluster refreshes will put things in a bad state constantly, which would be an ongoing headache. I don't think we lose anything by doing that, right? Developing on the script itself is already tricky.

**Comment by zarsky-broad** [comment]
My two concerns are:

1. How do we tell it where dev is? Add a config key?
2. What about prod? I want to make sure it's using it's own version.

**Comment by panentheos** [comment]
A config key (for both dev and prod) would be the obvious choice. It's ever so slightly brittle in that we will need to keep it updated when we change the deployment url, but that's not something that should be happening frequently. Thoughts?

**Comment by zarsky-broad** [comment]
Really I was just wondering if you had some magical alternate idea. My only thought is to only have the key on dev, and have prod fall through to window.location

**Pull request merged**